Modules/Core/Common/include/itkObjectFactoryBase.h

```diff
@@ -218,12 +218,10 @@ class ITKCommon_EXPORT ObjectFactoryBase : public Object
     {};
 
     // Factory registration, made thread-safe by "magic statics" (as introduced with C++11).
-    static const FactoryRegistration staticFactoryRegistration = [] {
+    [[maybe_unused]] static const FactoryRegistration staticFactoryRegistration = [] {
       RegisterFactoryInternal(TFactory::New());
       return FactoryRegistration{};
     }();
-
-    (void)staticFactoryRegistration;
```

Comment:
This may be a case ( or a previous implementation of this ) or something similar where the compiler may have optimized away the variable used to hold results of initialization, but that variable is never directly accessed.